### PR TITLE
integrating elk into mhn

### PR DESCRIPTION
**Pull request by aabed**

### CODE REVIEW

File: scripts/install_elk.sh [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+sudo apt-get update
+sudo apt-get -y install oracle-java8-installer
+wget -O - http://packages.elasticsearch.org/GPG-KEY-elasticsearch | sudo apt-key add -
+echo 'deb http://packages.elasticsearch.org/elasticsearch/1.4/debian stable main' | sudo tee /etc/apt/sources.list.d/elasticsearch.list
+sudo apt-get update
+sudo apt-get -y install elasticsearch=1.4.4
+sed -i '/network.host/c\network.host\:\ localhost' /etc/elasticsearch/elasticsearch.yml
+sudo service elasticsearch restart
+sudo update-rc.d elasticsearch defaults 95 10
+cd ~; wget https://download.elasticsearch.org/kibana/kibana/kibana-4.0.1-linux-x64.tar.gz
```
Comment (jt6211) [inline scripts/install_elk.sh]: can you make a temp dir and download everything into it?  the rm -rf ~/kibana-4* worries me.

File: scripts/install_elk.sh [hunk 2]
```diff
@@ -0,0 +1,88 @@
+sudo add-apt-repository -y ppa:webupd8team/java
+sudo apt-get update
+sudo apt-get -y install oracle-java8-installer
+wget -O - http://packages.elasticsearch.org/GPG-KEY-elasticsearch | sudo apt-key add -
+echo 'deb http://packages.elasticsearch.org/elasticsearch/1.4/debian stable main' | sudo tee /etc/apt/sources.list.d/elasticsearch.list
+sudo apt-get update
+sudo apt-get -y install elasticsearch=1.4.4
```
Comment (jt6211) [inline scripts/install_elk.sh]: Can you remove all the sudo's?  It looks like there are a mixture of commands where some assume root access and some use sudo.  I think the entire script should just be run as root to keep it simple.

File: scripts/install_elk.sh [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+cd /opt/logstash
+git clone https://github.com/aabed/logstash-input-hpfeeds.git
+echo "gem \"logstash-input-hpfeeds\", :path => \"/opt/logstash/logstash-input-hpfeeds\"" >> Gemfile
+
+bin/plugin install --no-verify
+#patching hpfeeds library to work with jruby as the bool function is not implemented yet and it doesn't affect the flow
+sed -ie '/Socket::Option.bool/ s/^#*/#/' /opt/logstash/vendor/bundle/jruby/1.9/gems/hpfeeds-0.1.6/lib/hpfeeds/client.rb
+
+SECRET=`python -c 'import uuid;print str(uuid.uuid4()).replace("-","")'`
+/opt/hpfeeds/env/bin/python /opt/hpfeeds/broker/add_user.py elk $SECRET "geoloc.events" amun.events,dionaea.connections,dionaea.capture,glastopf.events,beeswarm.hive,kippo.sessions,conpot.events,snort.alerts,kippo.alerts,wordpot.events,shockpot.events,p0f.events,suricata.events,elastichoney.events
```
Comment (jt6211) [inline scripts/install_elk.sh]: does this really need publish permissions for `geoloc.events`?

File: scripts/install_elk.sh [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+sudo apt-get update
+sudo apt-get -y install elasticsearch=1.4.4
+sed -i '/network.host/c\network.host\:\ localhost' /etc/elasticsearch/elasticsearch.yml
+sudo service elasticsearch restart
+sudo update-rc.d elasticsearch defaults 95 10
+cd ~; wget https://download.elasticsearch.org/kibana/kibana/kibana-4.0.1-linux-x64.tar.gz
+tar xvf kibana-*.tar.gz
+sed -i '/0.0.0.0/c\host\:\ localhost' /etc/elasticsearch/elasticsearch.yml
+sudo mkdir -p /opt/kibana
+sudo cp -R ~/kibana-4*/* /opt/kibana/
```
Comment (jt6211) [inline scripts/install_elk.sh]: Any chance you have a Kibana dashboard or multiple you could share?  I have a script that will dump and restore them.

**Comment by jt6211** [comment]
This looks great.  Can you make the changes I mentioned and I will merge this.  Thanks @aabed

**Comment by aabed** [comment]
working on changes

File: scripts/install_elk.sh (hunk 4 above)
Comment (aabed) [inline scripts/install_elk.sh]: I am afraid that I hadn't created any, yet
but let's publish this and hope that I will have time to create one, or maybe someone here will create and share one

**Comment by aabed** [comment]
ping

**Comment by jt6211** [comment]
LGTM

**Comment by jt6211** [comment]
Thanks @aabed

**Pull request merged**